core/shared/src/main/scala/coursier/util/Print.scala

```diff
@@ -4,6 +4,35 @@ import coursier.core.{ Attributes, Dependency, Module, Orders, Project, Resoluti
 
 object Print {
 
+  object Colors {
+    private val `with`: Colors = Colors(Console.RED, Console.YELLOW, Console.RESET)
+    private val `without`: Colors = Colors("", "", "")
+
+    def get(colors: Boolean): Colors = if (colors) `with` else `without`
+  }
+
+  case class Colors private(red: String, yellow: String, reset: String)
+
+  trait Renderable {
+    def repr(colors: Colors): String
+  }
+
+  trait Elem extends Renderable {
+    def dep: Dependency
+    def excluded: Boolean
+    def reconciledVersion: String
+    def children: Seq[Elem]
+  }
+
+  trait Parent extends Renderable {
+    def module: Module
+    def version: String
+    def dependsOn: Module
+    def wantVersion: String
+    def gotVersion: String
```

Comment:
Thanks for refactoring these out.

Q: What is the difference between `reconciledVersion` in `Elem` and `gotVersion` in `Parent`?

Reply:
Actually, nothing since I don't have to maintain backward compatibility - I'll change it.